query-service-impl/src/main/java/org/hypertrace/core/query/service/prometheus/PromQLRangeQuery.java

```diff
@@ -0,0 +1,21 @@
+package org.hypertrace.core.query.service.prometheus;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.List;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.Singular;
+
+@Getter
+@Builder
+class PromQLRangeQuery {
+  @NonNull @Singular private List<String> queries;
+
+  @NonNull private Instant startTime;
+
+  @NonNull private Instant endTime;
+
+  @NonNull private Duration step;
```

Comment:
Let's use consistent naming through our code. I think we call this period (maybe?) elsewhere.

Reply:
Have used the name as per the promql HTTP API - https://prometheus.io/docs/prometheus/latest/querying/api/#range-queries - for ease of co-relating. we can change it to a period with comments.

Reply:
Yeah I guess it's a bit challenging because we want consistency in our code but also consistency with their API. I guess since this is inside the promql impl details, calling it step might be fine with a comment to period, or the reverse as you suggested is fine too. I'd probably go with whichever results in more net readability - on the assignment into this class and on the read from this class when forming the http query param.